### ISSUE TITLE
Incremental builds

### ISSUE THREAD

**Issue by larsbrinkhoff** [PDP-10/its#187]
Invent some scheme to build things piecemeal.

**Comment by everyfan** [comment]
In ITS, we have a text file. Each line in text file consists of a program name, some white space, and then a version number, describing the current built version. In the git, we have a similar file, describing the current version in the git.

We use pexpect to extract the ITS file, use Python to parse it, and compare the version numbers. For each program that ITS has an earlier version of, we run a separate build script in pexpect.

Users can prevent a build by just using a really high version number. We don't necessarily need to have an accurate version number, we can start at 1 in the current build go up

**Comment by rcornwell** [comment]
Only thing I would recommend is not keeping it unix centric. Since many run simH on windows and other OS's

**Comment by everyfan** [comment]
That shouldn't be too hard; python has lots of platform-independent features

**Comment by larsbrinkhoff** [comment]
I'm not overly happy about the TCL-based build scripts.  Maybe something in Python would be better.  But they work, and it's a large effort to make something else.

**Comment by larsbrinkhoff** [comment]
There was another idea floating around.  The most of the build is spent waiting for console output, so the host CPU isn't very busy.  If the build could be scripted with xfiles and xxfiles instead, it could probably run at near full speed.  This would also be a large change.

**Comment by eswenson1** [comment]
Yes, but we could do some of this incrementally -- use the current TCL method, but rather than sending all the individual command lines to ITS, we could lump a whole bunch of them together in an XXFILE script and just run these XXFILES.  We could evolve to shorter and shorter TCL scripts, eventually doing away with most of them.
